src/serac/numerics/functional/domain.cpp

```diff
@@ -535,6 +535,49 @@ void Domain::insert_restriction(const serac::fes_t* fes, FunctionSpace space)
 
 const BlockElementRestriction& Domain::get_restriction(FunctionSpace space) { return restriction_operators.at(space); };
 
+void Domain::insert_shared_interior_face_list()
+{
+  // Weights only need to be computed for Domain of InteriorFaces type
+  SLIC_ERROR_IF(type_ != Domain::Type::InteriorFaces, "This method is only for interior face domains");
+
+  // make a list if we don't already have one
+  if (shared_interior_face_ids_.size() == 0) {
```

Comment:
I prefer this, as the intent is clearer:

```suggestion
  if (shared_interior_face_ids_.empty()) {
```